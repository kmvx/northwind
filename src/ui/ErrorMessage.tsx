import React from 'react';
import axios from 'axios';
import clsx from 'clsx';

export default function ErrorMessage({
  error,
  className,
}: {
  error: Error;
  className?: string;
}): React.JSX.Element {
  return (
    <div className={clsx('m-2 text-danger', className)}>
      <div>Error: {error.message}</div>
      {error instanceof axios.AxiosError && error?.response?.data && (
        <>
          <div>Data:</div>
          <div style={{ whiteSpace: 'pre-wrap' }}>
            {JSON.stringify(error?.response?.data, null, 4)}
          </div>
        </>
      )}
    </div>
  );
}
