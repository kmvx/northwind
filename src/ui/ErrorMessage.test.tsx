import { expect, test } from 'vitest';
import { render } from '@testing-library/react';
import ErrorMessage from './ErrorMessage';

test('ErrorMessage', async () => {
  const { container } = render(<ErrorMessage error={new Error('Test')} />);
  expect(container).toMatchSnapshot();
});
