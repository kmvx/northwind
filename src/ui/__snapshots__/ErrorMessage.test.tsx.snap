// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage 1`] = `
<div
  className="m-2 text-danger"
>
  <div>
    Error: 
    Test
  </div>
</div>
`;
